### PR TITLE
parser,ddl: data type default values support `REPLACE(UPPER(uuid()), '-', '')`

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/50937

Problem Summary:
* Data type default values support `REPLACE(UPPER(uuid()), '-', '')`
* When executing `show create table`, the expression becomes `REPLACE(CONVERT(UPPER(UUID()) USING UTF8MB4), '-', ''))`. So we support `REPLACE(CONVERT(UPPER(UUID()) USING UTF8MB4), '-', ''))` too.
  * Currently, the `show create table` function with the `convert` function is not supported.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/parser/parser.y [hunk 1]
```diff
@@ -3959,6 +3959,13 @@ BuiltinFunction:
 			Args:   $3.([]ast.ExprNode),
 		}
 	}
+|	"REPLACE" '(' ExpressionListOpt ')'
```
Comment (CbcWestwolf) [inline pkg/parser/parser.y:3962]: Can we remove this derivation and use the above derivation (`identifier '(' ExpressionList ')'`)?

Comment (zimulala) [reply pkg/parser/parser.y]: Done.

**Review by CbcWestwolf — APPROVED** [review]

File: pkg/ddl/db_integration_test.go [hunk 2]
```diff
@@ -1606,6 +1607,74 @@ func TestDefaultColumnWithRand(t *testing.T) {
 	tk.MustGetErrCode("CREATE TABLE t3 (c int, c1 int default a_function_not_supported_yet());", errno.ErrDefValGeneratedNamedFunctionIsNotAllowed)
 }
 
+func TestDefaultColumnWithReplace(t *testing.T) {
```
Comment (tangenta) [inline pkg/ddl/db_integration_test.go:1610]: Maybe we should move this to functional tests?

Comment (zimulala) [reply pkg/ddl/db_integration_test.go:1610]: There's an inconclusive inspection test here, so I didn't put it there.

**Pull request merged**